File: admin/types/restaurant.ts
export interface MenuItem {
  item: string;
  recommendation_level: 'must_try' | 'recommended' | 'mentioned';
}

export interface Location {
  city: string | null;
  neighborhood: string | null;
  address: string | null;
  region: 'North' | 'Center' | 'South' | null;
}

export interface ContactInfo {
  hours: string | null;
  phone: string | null;
  website: string | null;
}

export interface EpisodeInfo {
  video_id: string;
  video_url: string;
  title: string;
  channel_name: string;
  analysis_date: string;
}

export interface MentionTimestamp {
  timestamp: number;
  context: string;
}

export interface GooglePlacesInfo {
  place_id: string;
  name: string;
  formatted_address: string;
  rating: number;
  user_ratings_total: number;
  price_level: number;
  photos: string[];
}

export interface Restaurant {
  id?: string;
  name_hebrew: string;
  name_english: string | null;
  location: Location;
  cuisine_type: string;
  status: 'open' | 'closed' | 'new_opening' | 'closing_soon' | 'reopening';
  price_range: 'budget' | 'mid-range' | 'expensive' | 'not_mentioned';
  host_opinion: 'positive' | 'negative' | 'mixed' | 'neutral';
  host_comments: string;
  menu_items: MenuItem[];
  special_features: string[];
  contact_info: ContactInfo;
  business_news: string | null;
  mention_context: 'new_opening' | 'review' | 'news' | 'recommendation' | 'comparison' | 'business_news';
  episode_info?: EpisodeInfo;
  mention_timestamps?: MentionTimestamp[];
  google_places?: GooglePlacesInfo;
  rating?: {
    google_rating: number;
    total_reviews: number;
    price_level: number;
  };
  food_trends?: string[];
  created_at?: string;
  updated_at?: string;
}

export interface RestaurantListResponse {
  restaurants: Restaurant[];
  pagination: {
    page: number;
    limit: number;
    total: number;
    totalPages: number;
  };
}

export interface EditHistory {
  id: string;
  restaurant_id: string;
  restaurant_name: string;
  admin_email: string;
  admin_name: string;
  edit_type: 'create' | 'update' | 'delete' | 'approve' | 'reject';
  changes: Record<string, any>;
  timestamp: string;
}

File: api/models/restaurant.py
"""Pydantic models for restaurant-related data."""

from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field
from datetime import datetime


class Location(BaseModel):
    """Restaurant location details."""
    city: Optional[str] = None
    address: Optional[str] = None
    region: Optional[str] = None
    lat: Optional[float] = None
    lng: Optional[float] = None


class Rating(BaseModel):
    """Restaurant rating information."""
    google_rating: Optional[float] = None
    review_count: Optional[int] = None


class EpisodeInfo(BaseModel):
    """Information about the episode where restaurant was mentioned."""
    video_id: Optional[str] = None
    video_url: Optional[str] = None
    analysis_date: Optional[str] = None


class Restaurant(BaseModel):
    """Full restaurant model."""
    id: Optional[str] = None
    name_hebrew: str
    name_english: Optional[str] = None
    cuisine_type: Optional[str] = None
    location: Optional[Location] = None
    price_range: Optional[str] = None
    status: Optional[str] = None
    host_opinion: Optional[str] = None
    rating: Optional[Rating] = None
    episode_info: Optional[EpisodeInfo] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None

    class Config:
        extra = "allow"  # Allow additional fields


class RestaurantCreate(BaseModel):
    """Model for creating a restaurant."""
    name_hebrew: str
    name_english: Optional[str] = None
    cuisine_type: Optional[str] = None
    location: Optional[Location] = None
    price_range: Optional[str] = None
    status: Optional[str] = None
    host_opinion: Optional[str] = None

    class Config:
        extra = "allow"


class RestaurantUpdate(BaseModel):
    """Model for updating a restaurant."""
    name_hebrew: Optional[str] = None
    name_english: Optional[str] = None
    cuisine_type: Optional[str] = None
    location: Optional[Location] = None
    price_range: Optional[str] = None
    status: Optional[str] = None
    host_opinion: Optional[str] = None

    class Config:
        extra = "allow"


class RestaurantList(BaseModel):
    """Response model for restaurant list."""
    restaurants: List[Restaurant]
    count: int


class RestaurantSearchResponse(BaseModel):
    """Response model for restaurant search."""
    restaurants: List[Restaurant]
    timeline_data: Optional[List[Dict[str, Any]]] = None
    analytics: Optional[Dict[str, Any]] = None


class Pagination(BaseModel):
    """Pagination metadata."""
    page: int
    limit: int
    total: int
    totalPages: int


class PaginatedRestaurants(BaseModel):
    """Paginated restaurant response."""
    restaurants: List[Restaurant]
    pagination: Pagination
